File: editor/src/de/project/ice/editor/undoredo/RemoveEntityAction.java
package de.project.ice.editor.undoredo;

import com.badlogic.ashley.core.Engine;
import com.badlogic.ashley.core.Entity;

public class RemoveEntityAction implements UndoableRedoableAction
{
    private Entity entity;
    private Engine engine;

    public RemoveEntityAction(Entity entity, Engine engine)
    {
        this.entity = entity;
        this.engine = engine;
    }


    @Override
    public void undo()
    {
        engine.addEntity(entity);
    }

    @Override
    public void redo()
    {
        engine.removeEntity(entity);
    }

}

File: editor/src/de/project/ice/editor/undoredo/RemoveEntityAction.kt
package de.project.ice.editor.undoredo

import com.badlogic.ashley.core.Engine
import com.badlogic.ashley.core.Entity

class RemoveEntityAction(private val entity: Entity, private val engine: Engine) : UndoableRedoableAction {

    override fun undo() {
        engine.addEntity(entity)
    }

    override fun redo() {
        engine.removeEntity(entity)
    }

}
